src/sql/workbench/browser/modelComponents/tree.component.ts

```diff
@@ -119,6 +121,15 @@ export default class TreeComponent extends ComponentBase implements IComponent,
 			this._register(this._tree.onDidChangeSelection(e => {
 				this._dataProvider.onNodeSelected(e.selection);
 			}));
+			this.onkeydown(this._inputContainer.nativeElement, (e: StandardKeyboardEvent) => {
+				// Enter on a tree will select the currently selected item as the default behavior
+				// but if not stopped here then will propagate up.
+				// This might have unintended effects such as a dialog closing.
+				if (e.keyCode === KeyCode.Enter) {
+					this._tree.toggleExpansion(this._tree.getFocus());
```

Comment:
toggleExpansion... does that mean we will close too if its already opened, I don't have any specific case against that but might be bit confusing to user. What do you say?

Reply:
Yup. That's the current behavior for OE and seems like the correct way to handle that to me. Easy change if we think just opening is the correct way - I'll run it by the accessibility team.